scripts/sentiment_analysis/sentiment_analysis_cnn.py

```diff
@@ -0,0 +1,319 @@
+    pool = mp.Pool(2)
+    dataset = gluon.data.SimpleDataset(pool.map(preprocess, dataset))
+    lengths = gluon.data.SimpleDataset(pool.map(get_length, dataset))
+    end = time.time()
+    print('Done! Tokenizing Time={:.2f}s, #Sentences={}'.format(end - start, len(dataset)))
+    return dataset, lengths
+
+# Preprocess the dataset
+train_dataset, train_data_lengths = preprocess_dataset(train_dataset)
+valid_dataset, valid_data_lengths = preprocess_dataset(valid_dataset)
+test_dataset, test_data_lengths = preprocess_dataset(test_dataset)
+
+# Construct the DataLoader. Pad data and stack label
+batchify_fn = nlp.data.batchify.Tuple(nlp.data.batchify.Pad(axis=0, ret_length=True),
+                                      nlp.data.batchify.Stack(dtype='float32'))
+if args.bucket_type is None:
+    print('Bucketing strategy is not used!')
+    train_dataloader = DataLoader(dataset=train_dataset,
+                                  batch_size=args.batch_size,
+                                  shuffle=True,
+                                  batchify_fn=batchify_fn)
+else:
+    if args.bucket_type == 'fixed':
+        print('Use FixedBucketSampler')
+        batch_sampler = nlp.data.FixedBucketSampler(train_data_lengths,
+                                                    batch_size=args.batch_size,
+                                                    num_buckets=args.bucket_num,
+                                                    ratio=args.bucket_ratio,
+                                                    shuffle=True)
+        print(batch_sampler.stats())
+    elif args.bucket_type == 'sorted':
+        print('Use SortedBucketSampler')
+        batch_sampler = nlp.data.SortedBucketSampler(train_data_lengths,
+                                                     batch_size=args.batch_size,
+                                                     mult=args.bucket_mult,
+                                                     shuffle=True)
+    else:
+        raise NotImplementedError
+    train_dataloader = DataLoader(dataset=train_dataset,
+                                  batch_sampler=batch_sampler,
+                                  batchify_fn=batchify_fn)
+
+valid_dataloader = DataLoader(dataset=valid_dataset,
+                              batch_size=args.batch_size,
+                              shuffle=False,
+                              sampler=nlp.data.SortedSampler(valid_data_lengths),
+                              batchify_fn=batchify_fn)
+
+test_dataloader = DataLoader(dataset=test_dataset,
+                             batch_size=args.batch_size,
+                             shuffle=False,
+                             sampler=nlp.data.SortedSampler(test_data_lengths),
+                             batchify_fn=batchify_fn)
+
+
+net.hybridize()
+print(net)
+
+net.initialize(mx.init.Xavier(), ctx=context)
+net.embedding.weight.set_data(vocab.embedding.idx_to_vec)
+net.embedding_extend.weight.set_data(vocab.embedding.idx_to_vec)
```

Comment:
set for embedding_extend only if self._use_multichannel. 
Thus, this should depend on `if args.model_mode == xxx ...`